includes/paragraphs.inc

```diff
@@ -377,5 +373,124 @@ function get_bcl_variant(Paragraph $paragraph): string {
+  // Overwrite the variant because is an image.
+  $variables['variant'] = 'image';
+  $cacheability->applyTo($variables);
+
+  return $variables;
+}
+
+/**
+ * Set the embedded media.
+ *
+ * @param bool $is_image
+ *   Flag to determine if is an image or not.
+ * @param array $variables
+ *   Drupal variables.
+ * @param Drupal\media\Entity\Media $media
+ *   Media object.
+ * @param Drupal\Core\Cache\CacheableMetadata $cacheability
+ *   CacheableMetadata object.
+ * @param Drupal\media\MediaSourceInterface $source
+ *   Media source.
+ *
+ * @return array
+ *   Updated variables.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function set_embedded_media(bool $is_image, array &$variables, Media $media, CacheableMetadata $cacheability, MediaSourceInterface $source): array {
+  if ($is_image) {
+    return $variables;
+  }
+
+  // Default video aspect ratio is set to 16x9.
+  $variables['ratio'] = '16x9';
+
+  // Load information about the media and the display.
+  $media_type = \Drupal::entityTypeManager()->getStorage('media_type')->load($media->bundle());
+  $cacheability->addCacheableDependency($media_type);
+  $source_field = $source->getSourceFieldDefinition($media_type);
+  $display = EntityViewDisplay::collectRenderDisplay($media, 'oe_theme_main_content');
+  $cacheability->addCacheableDependency($display);
+  $display_options = $display->getComponent($source_field->getName());
+
+  // If it is an OEmbed resource, render it and pass it as embeddable data
+  // only if it is of type video or html.
+  if ($source instanceof OEmbed) {
+    $oembed_type = $source->getMetadata($media, 'type');
+    if (in_array($oembed_type, ['video', 'html'])) {
+      $variables['embedded_media'] = $media->{$source_field->getName()}->view($display_options);
+      $cacheability->applyTo($variables);
+      return $variables;
+    }
+  }
+
+  // If its an AvPortal video or an iframe video, render it.
+  $variables['embedded_media'] = $media->{$source_field->getName()}->view($display_options);
+  $cacheability->applyTo($variables);
+  set_ratio($media, $variables, $cacheability);
+  return $variables;
+}
+
+/**
+ * Set the video iframe ratio.
+ *
+ * @param Drupal\media\Entity\Media $media
+ *   Media object.
+ * @param array $variables
+ *   Drupal variables.
+ * @param Drupal\Core\Cache\CacheableMetadata $cacheability
+ *   CacheableMetadata object.
+ *
+ * @return array
+ *   Updated variables.
+ */
+function set_ratio(Media $media, array &$variables, CacheableMetadata $cacheability): array {
+  // When dealing with iframe videos, also respect its given aspect ratio.
+  if ($media->bundle() !== 'video_iframe') {
+    return $variables;
+  }
+  $ratio = $media->get('oe_media_iframe_ratio')->value;
+  $variables['ratio'] = str_replace('_', 'x', $ratio);
+  $cacheability->applyTo($variables);
+  return $variables;
```

Comment:
Is needed because of the sign of the method: 

`function set_video_iframe_ratio(Media $media, array &$variables, CacheableMetadata $cacheability): array {`

is waiting an array as the output of the method. So the return is needed for this reason.